packages/experiments/src/utilities/keysequence/IKeytipTransitionKey.ts

```diff
@@ -0,0 +1,65 @@
+import { KeytipTransitionModifier } from '../../Keytip';
+
+export interface IKeytipTransitionKey {
+  key: string;
+  modifierKeys?: KeytipTransitionModifier[];
+}
+
+/**
+ * Tests for equality between two IKeytipTransitionKeys
+ *
+ * @param key1 - First IKeytipTransitionKey
+ * @param key2 - Second IKeytipTransitionKey
+ * @returns {boolean} T/F if the transition keys are equal
+ */
+export function transitionKeysAreEqual(key1: IKeytipTransitionKey, key2: IKeytipTransitionKey): boolean {
+  if (key1.key !== key2.key) {
+    return false;
+  }
+
+  let mod1 = key1.modifierKeys;
+  let mod2 = key2.modifierKeys;
+
+  if ((!mod1 && mod2) || (mod1 && !mod2)) {
+    // Not equal if one modifier is defined and the other isn't
+    return false;
+  }
+
+  if (mod1 && mod2) {
+    if (mod1.length !== mod2.length) {
+      return false;
+    }
+
+    // Check that all items in mod1 are in mod2
+    // Order doesn't matter
+    let mod2Copy = [...mod2];
+    for (let i = 0; i < mod1.length; i++) {
+      let index = mod2Copy.indexOf(mod1[i]);
+      if (index < 0) {
+        return false;
+      }
+      mod2Copy.splice(index, 1);
+    }
+    if (mod2Copy.length !== 0) {
+      return false;
+    }
+  }
```

Comment:
This is N^2, can we use a JavaScript set instead? I'm not sure what the policy on using Set is in fabric, it is supported by IE11, but it is in typescripts ES2015.collections library, which is not included by default when transpiling to ES5. Thoughts @dzearing ?

Reply:
What do you think about sorting the two arrays and looping through? The arrays are just of enums (so numbers) so they can be sorted naturally

Reply:
How big do you expect these arrays to get? Sorting will still be slower than using the set, which might matter if these lists get pretty long or this code runs a lot.

Reply:
The arrays are just a list of modifiers (Alt, Shift, Ctrl). We only have 5 defined now so that would be the max, but the developer wouldn't define more then 2-3 (otherwise the user has to press 5 keys to enable keytip mode)

Reply:
Sorting is fine then